include/deal.II/dofs/dof_tools.h

```diff
@@ -1150,10 +1150,36 @@ namespace DoFTools
    * This function makes sure that identity constraints don't create cycles
    * in @p constraints.
    *
+   * @p periodicity_factor can be used to to implement Bloch periodic conditions
+   * (a.k.a. phase shift periodic conditions) of the form
+   * $\psi(\mathbf{r})=e^{-i\mathbf{k}\cdot\mathbf{r}}u(\mathbf{r})$
+   * where $u$ is periodic with the same periodicity as the crystal lattice and
+   * and $k$ is the wavevector, see
+   * [Photonic Crystals: Molding the Flow of Light by J.D. Joannopoulos et
+   * al](https://press.princeton.edu/books/hardcover/9780691124568/photonic-crystals).
```

Comment:
I would consider referring to https://en.wikipedia.org/wiki/Bloch_wave instead since that is freely available.